libs/vision/include/mrpt/vision/pnp_algos.h

```diff
@@ -47,47 +47,43 @@ namespace mrpt
             {
                 public:
                 
-                    #if MRPT_HAS_OPENCV
-                    
-                        /**
-                         * @brief \cite hesch Direct Least Squares (DLS) - PnP : Algorithm formulates position as a function of rotation. 
-                         *        Use Cayley's rotation theorem to represent the rotation using parameters ($s_1, s_2, s_3$).
-                         *        Solve the rotation using multi-variate polynomial expressions
-                         *        
-                         * @param[in] obj_pts Object points in Camera Co-ordinate system {C} nX3 (only 4 points used) array [p_x, p_y, p_z]
-                         * @param[in] img_pts Image points in pixels nX3 (only 4 points used) array containing pixel data from camera [u, v, 1]
-                         * @param[in] n number of 2D-3D correspondences
-                         * @param[in] cam_intrinsic Camera Intrinsic matrix
-                         * @param[out] pose_mat Output pose vector 6X1, pose_mat[0:2]-> Translation, pose_mat[3:5] -> Quaternion vector component 
-                         * @return success flag
-                         */
-                        int dls(const Eigen::Ref<Eigen::MatrixXd> obj_pts, const Eigen::Ref<Eigen::MatrixXd> img_pts, int n, const Eigen::Ref<Eigen::MatrixXd> cam_intrinsic, Eigen::Ref<Eigen::MatrixXd> pose_mat);
+                    /**
+                     * @brief \cite hesch Direct Least Squares (DLS) - PnP : Algorithm formulates position as a function of rotation. 
+                     *        Use Cayley's rotation theorem to represent the rotation using parameters ($s_1, s_2, s_3$).
+                     *        Solve the rotation using multi-variate polynomial expressions
+                     *        
+                     * @param[in] obj_pts Object points in Camera Co-ordinate system {C} nX3 (only 4 points used) array [p_x, p_y, p_z]
+                     * @param[in] img_pts Image points in pixels nX3 (only 4 points used) array containing pixel data from camera [u, v, 1]
+                     * @param[in] n number of 2D-3D correspondences
+                     * @param[in] cam_intrinsic Camera Intrinsic matrix
+                     * @param[out] pose_mat Output pose vector 6X1, pose_mat[0:2]-> Translation, pose_mat[3:5] -> Quaternion vector component 
+                     * @return success flag
+                     */
+                    int dls(const Eigen::Ref<Eigen::MatrixXd> obj_pts, const Eigen::Ref<Eigen::MatrixXd> img_pts, int n, const Eigen::Ref<Eigen::MatrixXd> cam_intrinsic, Eigen::Ref<Eigen::MatrixXd> pose_mat);
```

Comment:
I know this is a bit of "style flame", but... I think that returning an `int` as a success flag is very "C'ish", right?? :-) 
If there only two possible return values from all these new functions, I would advice switching to `bool`. The more widespread convention throughout MRPT is that returning false means `error`. 

However, I saw that, at least in some cases, the function always return "1" except in the case of not being compiled against OpenCV... such situations should (as said above) raise an exception, because is an excepcionally important error that should not undergo unnoticed by the user. So, if the function can only return "1" always... I would make it `void`, with the option to launch an exception if not available! 
Simpler is better! ;-) 

(cc: @jbriales, @famoreno )